src/advanced/src/components/OrderSummary.tsx

```diff
@@ -0,0 +1,155 @@
+import React from 'react';
+import { CartState } from '../types/cart';
+import { Product } from '../types/product';
+import { calcCartTotals, calcBonusPoints } from '../utils/calculation';
+
+interface OrderSummaryProps {
+  cartState: CartState;
+  productList: Product[];
+}
+
+const OrderSummary: React.FC<OrderSummaryProps> = ({
+  cartState,
+  productList,
+}) => {
+  const totals = calcCartTotals(cartState.cartItems, productList, new Date());
+  const bonus = calcBonusPoints(cartState.cartItems, productList, new Date());
```

Comment:
today 를 미리 변수로 뽑아도 좋을 것 같고 args 로 만들어서 넘겨도 좋을 것 같아유
```suggestion
  const today = new Date();
  const cartArgs = [cartState.cartItems, productList, today];
  const [totals, bonus] = [calcCartTotals(...cartArgs), calcBonusPoints(...cartArgs)];
```

근데 쓰고보니 그냥 유현님 코드가 더 가독성 좋은 것 같아요 ㅋㅋㅋㅋㅋㅋ